app/dependencies/redis_client.py

```diff
@@ -3,25 +3,87 @@
 from datetime import timedelta
 
 import redis
+from dotenv import load_dotenv
 
-redis_client = redis.Redis(
-    host=os.getenv("REDIS_HOST"), port=os.getenv("REDIS_PORT"), db=0
-)
+# Load environment variables from .env file
+load_dotenv()
 
+redis_client = redis.Redis(host=os.environ.get("REDIS_HOST"), port=os.environ.get("REDIS_PORT"), db=0)
 
-def get_redis_key():
-    return "hex_data"
 
-
-def set_redis_data(data):
-    key = get_redis_key()
+def set_redis_data(key, data):
     redis_client.set(key, json.dumps(data))
     redis_client.expire(key, timedelta(days=1))
 
 
-def get_redis_data():
-    key = get_redis_key()
+def get_hexdata_from_redis(key):
     data = redis_client.get(key)
     if data:
-        return json.loads(data)
+        # str_data = data.decode('utf-8')
+        json_data = json.loads(data)
+
+        cryptocurrencies = []
+        symbols_c = ""
+        # Iterate through the `data` list
+        for crypto in json_data["data"]:
+            # Extract the `name` and `symbol` of each cryptocurrency
+            name = crypto["name"]
+            symbol = crypto["symbol"]
+            id = str(crypto["id"])
+            price = crypto["quote"]["USD"]["price"]  # Extract price
+
+            symbols_c += symbol + ","
+            # Append the `name` and `symbol` to the `cryptocurrencies` list as a dictionary
+            cryptocurrencies.append(
+                {"name": name, "id": id, "symbol": symbol, "price": price}
+            )
+
+        return cryptocurrencies
+
+    return None
+
+
+def get_historical_redis_data(key):
+    data = redis_client.get(key)
+    if data is not None:
+        json_data = json.loads(data)
+
+        return json_data
+    else:
+        return None
+
+
+def get_currencylist_redis_data(key):
+    data = redis_client.get(key)
```

Comment:
this can be refactored like

data = redis_client.get(key)
if data:
   return json.loads(data)
   
 return None